src/Microsoft.IdentityModel.JsonWebTokens/JsonWebTokenHandler.cs

```diff
@@ -530,6 +701,9 @@ private string EncryptTokenPrivate(string innerJwt, EncryptingCredentials encryp
 
                 header.Add(JwtHeaderParameterNames.Typ, JwtConstants.HeaderType);
 
+                if (additionalHeaderClaims != null)
```

Comment:
What happens if the algorithm is not "dir" (the _else_ part)?
It looks like we won't add additional header claims in that case.

Reply:
Good catch! I somehow missed the fact that we're creating the JWE header in two different places depending on the algorithm used. 

I fixed this and also added in some tests for both the direct encryption and key wrapping scenarios.